src/Forms/VocabForm/VocabForm.jsx

```diff
@@ -0,0 +1,337 @@
+import React, { useCallback, useState, useEffect, useContext } from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch, useSelector } from "react-redux";
+
+import Button from "../../Components/Button/Button.jsx";
+import ArrayTextInput from "../../Components/Form/ArrayTextInput/ArrayTextInput.jsx";
+import Select from "../../Components/Form/Select/Select.jsx";
+import Switch from "../../Components/Form/Switch/Switch.jsx";
+import TextInput from "../../Components/Form/TextInput/TextInput.jsx";
+import Modal from "../../Components/Modal/Modal.jsx";
+import GroupForm from "../../Forms/GroupForm/GroupForm.jsx";
+import PackageForm from "../../Forms/PackageForm/PackageForm.jsx";
+import SnackbarContext from "../../context/SnackbarContext.jsx";
+
+import { setVocabActive, setVocabActivate } from "../../redux/Actions/form.js";
+import { getPackages, createVocabulary } from "../../utils/api.js";
+import { languages, maxTranslations } from "../../utils/constants.js";
+
+const CustomSelectOption = ({ name, postfix }) => {
+  return (
+    <span className="custom-option-wrapper">
+      {name}
+      <small className="postfix">{postfix}</small>
+    </span>
+  );
+};
+
+const VocabForm = () => {
+  const { t } = useTranslation();
+
+  const { showSnack } = useContext(SnackbarContext);
+  const dispatch = useDispatch();
+
+  const active = useSelector((state) => state.form.vocab.active);
+  const activate = useSelector((state) => state.form.vocab.activate);
+
+  const [packages, setPackages] = useState([]);
+  const [packageItems, setPackageItems] = useState([]);
+  const [selectedPackage, setSelectedPackage] = useState(null);
+
+  const [groups, setGroups] = useState([]);
+  const [groupsItems, setGroupsItems] = useState([]);
+  const [selectedGroup, setSelectedGroup] = useState(null);
+
+  const [foreignWord, setForeignWord] = useState("");
+  const [translations, setTranslations] = useState([]);
+  const [description, setDescription] = useState("");
+
+  const [showAddPackage, setShowAddPackage] = useState(false);
+  const [showAddGroup, setShowAddGroup] = useState(false);
+
+  const onChangeActive = useCallback(() => {
+    dispatch(setVocabActive({ active: !active }));
+  }, [dispatch, active]);
+
+  const onChangeActivate = useCallback(() => {
+    dispatch(setVocabActivate({ activate: !activate }));
+  }, [dispatch, activate]);
+
+  const fetchPackages = useCallback(() => {
+    getPackages(true)
+      .then(({ data }) => {
+        setPackages(data);
+      })
+      .catch((err) => {
+        showSnack("error", t("global.fetchError"));
+      });
+  }, [showSnack, t]);
+
+  const onClear = useCallback(() => {
+    setSelectedPackage(null);
+    setSelectedGroup(null);
```

Comment:
I think it gives a better UX if we don't delete the package and group when you enter a vocab.